javaagent-instrumentation-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/InstrumentationContext.java

```diff
@@ -18,7 +18,12 @@ private InstrumentationContext() {}
    * <p>In reality, the <em>calls</em> to this method are re-written to something more performant
    * while injecting advice into a method.
    *
-   * <p>This method must only be called within an Advice class.
+   * <p>When this method is called from outside of an Advice class it can only access {@link
+   * ContextStore} when it is already created. For this {@link ContextStore} either needs to be
+   * added to {@code
+   * io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule#getMuzzleContextStoreClasses()}
+   * or be used in an Advice class which automatically adds it to {@code
+   * InstrumentationModule#getMuzzleContextStoreClasses()}
```

Comment:
The muzzle codegen plugin collects invocations from all helper classes, not only advices -- at least it should work that way.